kontrakter/felles/kodeverk/1.0/kodeverk.schema.json

```diff
@@ -0,0 +1,70 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://domstol.no/felles/1.0/kodeverk.schema.json",
+  "description": "Kodeverk som er definert i json filer i felles repo",
+
+  "type": "object",
+  "properties": {
+    "kodeType": {
+      "$ref": "#/definitions/kode",
+      "description": "unik id for denne typen koder."
+    },
+    "koder": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/kodeverkDefinisjon"
+      },
+      "minItems": 1
+    }
+  },
+  "required": ["kodeType"],
+  "additionalProperties": false,
+
+  "definitions": {
+    "kodeverkDefinisjon": {
+      "type": "object",
+      "description": "En kodeverk definisjon som brukes i definisjonen av kodeverk objekt",
+      "properties": {
+        "kode": {
+          "$ref": "#/definitions/kode"
+        },
+        "beskrivelse": {
+          "type": "string",
+          "maxLength": 100,
+          "minLength": 1
+        },
+        "gyldigTil": {
+          "type": "string",
+          "description": "Dato tid fordi felles med KDI og Microsoft liker ikke dato så da blir det 23.03.2023 00:00 betyr gyldig til og med 22.03.2023. Vi skal ikke ha med tidspunkt",
```

Comment:
Kan vi ikke ha med tid også? Så unngår vi hele KDI problematikken

Reply:
Du refererer til en eller annen "KDI problematikk" så derfor svarer jeg her nå i et forsøk på å legge den død. 

På generelt grunnlag så bør slike skjema være teknologi-agnostisk og man bør bruke ISO 8601 for representasjon av datoer og tid.

Json-schema spesifikasjonen har definert dette slik: 

![image](https://user-images.githubusercontent.com/131260037/233306056-c14bf58e-aa40-4152-893f-4c7e9ddeb219.png)

Og her er et eksempel-skjema:

```
{
    "$id": "https://example.com/datetimez/1.0/datetimez.schema.json",
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "description": "Testskjema for dato og dato-tid.",
    "title": "Datetimez",
    "type": "object",
    "properties": {
      "fraDatoTid": {
        "description": "Fra dato-tid, som domstolene spesifiserer. Eksempel: 2018-11-13T20:20:39+00:00",
        "type": "string",
        "format": "date-time"
      },
      "tilDato": {
        "description": "Til dato, som domstolene spesifiserer. Eksempel: 2018-11-13",
        "type": "string",
        "format": "date"
      },
      "tilTid": {
        "description": "Til TID, som domstolene spesifiserer. Eksempel: 20:20:39+00:00",
        "type": "string",
        "format": "time"
      }
    }
  } 
```

I praksis så er det slik at vi benytter ulike språk, teknologier og rammeverk på alle sider her som gjør at alle må forholde seg til mottak, serialisering og deserialisering av disse. Vi, altså KDI, benytter i stor grad microsoft dotnet og har da ikke noe spesifikt Date-objekt, som gjør at vi fort ender opp med å ta i mot "date" fra meldingen og behandle det som en date-time videre inn mot fagsystemet. Eller vi må legge på noe mapping eller lignende. Det blir vel fort samme utfordring for alle sider her, så selve skjema bør derfor etter min personlige mening være så teknologi-agnostisk som mulig og følge industri-standarder.

Reply:
Det er fint, men det er besluttet at KDI skal ha date-time i meldinger selv om det er dato ref. fødselsdato. Foreslår vi tar det opp på teknisk møte.

Reply:
@martinmyran Hvis KDI har endret mening og ønsker teknologi-agnostiske kontrakter så er jo det en win for alle parter? @JosteinKlakegg jobber jo i KDI.

Reply:
Vi bruker date som bestemt på teknisk koordineringsmøte.